polars/polars-lazy/polars-plan/src/logical_plan/optimizer/type_coercion/mod.rs

```diff
@@ -515,27 +515,28 @@ fn early_escape(type_self: &DataType, type_other: &DataType) -> Option<()> {
     }
 }
 
-#[cfg(test)]
```

Comment:
Can we fix the compilation error?

Reply:
I looked into it but it's not that simple. This test hasn't been run in ages and it shows.

The function being tested `optimize_expr` is no longer a standalone function. And to test it we apparently need to specify expr arena/node and lp_arena/node and I don't really know how that infrastructure works yet.

It's blocking other optimizations to the test suite so I'd rather get rid of it now and address it sometime next month when we can look at it together.

Reply:
Alright, can you open an issue for that one?

Reply:
See #9828